src/urllib3/connectionpool.py

```diff
@@ -56,10 +56,10 @@
 
 log = logging.getLogger(__name__)
 
-_Default = object()
+_Default: float = cast(float, object())
```

Comment:
Some other options:

- Mark the sentinels as `Any`, then you don't need the cast, but the default parameter assignment still works. And people would perhaps be less tempted to use them as actual floats.

- Add a real sentinel type (e.g. a one-item enum) and add it to the types and all. This is completely type safe but somewhat annoying.

I think all approaches are probably fine.

Reply:
Thanks, @bluetech for the review and suggestion.
Yeah, you are right. we have to add a real sentinel type (based on the discussion in the dev channel)